### PR TITLE
[bitnami/harbor] Feature/external db url secret

### PR DESCRIPTION
**Pull request by voor**
TL;DR
=====
- Allow Postgres endpoints to be defined via environment secretKeyRef
- Disable SSL by default (no longer `nil`) for external database
  connections

Detail
======
This change allows you to define the Postgres endpoint as a secret
value, rather than providing passwords in raw values files. A new secret
template is created for notary to introduce relevant environment
variables for the notary service.

Additionally, the default configuration for `externalDatabase.sslmode`
was configured as `nil`. This change explicitly sets `disable` for both
`values.yaml` and `values-production.yaml`.

Chart version received a patch version bump. This _could_ potentially
be seen as a minor version bump, depending on your interpretation of
semantic versioning.

Co-authored-by: @conzetti

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/harbor/templates/notary/notary-secret-envvars.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+    app.kubernetes.io/component: notary
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+type: Opaque
+data:
+  server_db_url: {{ include "harbor.database.notaryServer" . }}
```
Comment (FraPazGal) [inline bitnami/harbor/templates/notary/notary-secret-envvars.yaml:16]:
```suggestion
  server_db_url: {{ include "harbor.database.notaryServer" . | b64enc | quote }}
```

We could add a bit of encryption to the data by default.

Also, please add a newline at the end of the file.

**Review by FraPazGal — COMMENTED** [review]
Hello @voor,

Thank you very much for opening this PR, it looks good! Could you please take a look at my comments?

**Comment by voor** [comment]
@FraPazGal thank you for the review, added in all suggestions.

**Review by FraPazGal — APPROVED** [review]
LGTM!
